azure/scope/managedcontrolplane.go

```diff
@@ -731,18 +740,21 @@ func (s *ManagedControlPlaneScope) SetAnnotation(key, value string) {
 
 // TagsSpecs returns the tag specs for the ManagedControlPlane.
 func (s *ManagedControlPlaneScope) TagsSpecs() []azure.TagsSpec {
-	return []azure.TagsSpec{
-		{
-			Scope:      azure.ResourceGroupID(s.SubscriptionID(), s.ResourceGroup()),
-			Tags:       s.AdditionalTags(),
-			Annotation: azure.RGTagsLastAppliedAnnotation,
-		},
+	specs := []azure.TagsSpec{
 		{
 			Scope:      azure.ManagedClusterID(s.SubscriptionID(), s.ResourceGroup(), s.ManagedClusterSpec().ResourceName()),
 			Tags:       s.AdditionalTags(),
 			Annotation: azure.ManagedClusterTagsLastAppliedAnnotation,
 		},
 	}
+	if s.UseLegacyGroups {
+		specs = append(specs, azure.TagsSpec{
```

Comment:
If using legacy groups, would we need both Tags Specs? I may be misunderstanding here but it looks like we will have two copies of the additional tags here.

Reply:
One tags spec is for the resource group and one is for the managed cluster. With ASO, tags for a resource are reconciled alongside the rest of the resource instead of with the separate `tags` service which these serve as input to. So for an ASO resource, we don't need to define a tags spec.